broadleaf-thymeleaf3-presentation/src/main/java/org/broadleafcommerce/presentation/thymeleaf3/config/Thymeleaf3ConfigUtils.java

```diff
@@ -137,8 +141,8 @@ protected ITemplateResolver createCorrectTemplateResolver(BroadleafTemplateResol
         }
     }
     
-    protected ClassLoaderTemplateResolver createClassLoaderTemplateResolver(BroadleafTemplateResolver resolver) {
-        ClassLoaderTemplateResolver classpathResolver = new ClassLoaderTemplateResolver();
+    protected BroadleafClassLoaderTemplateResolver createClassLoaderTemplateResolver(BroadleafTemplateResolver resolver) {
+        BroadleafClassLoaderTemplateResolver classpathResolver = new BroadleafClassLoaderTemplateResolver(cacheIsTemplateExist);
```

Comment:
thinking about that a bit more, seems it makes more sense to simply check that cacheIsTemplateExist here and create either  BroadleafClassLoaderTemplateResolver  or  ClassLoaderTemplateResolver..